Description: Fix test failures with SQLite 3.37.0
 #   Failed test 'columns_info works'
 #   at t/60core.t line 390.
 #     Structures begin differing at:
 #          $got->{rank}{data_type} = 'INTEGER'
 #     $expected->{rank}{data_type} = 'integer'
 .
 Use Test::Deep to be able to use case-insensitive regexps.
Origin: vendor
Bug: https://github.com/Perl5/DBIx-Class/pull/141
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2022-01-18

--- a/t/60core.t
+++ b/t/60core.t
@@ -4,6 +4,7 @@
 use Test::More;
 use Test::Exception;
 use Test::Warn;
+use Test::Deep;
 use lib qw(t/lib);
 use DBICTest ':DiffSQL';
 
@@ -387,7 +388,7 @@
   $schema->source("Artist")->column_info_from_storage(1);
   $schema->source("Artist")->{_columns_info_loaded} = 0;
 
-  is_deeply (
+  cmp_deeply (
     $schema->source('Artist')->columns_info,
     {
       artistid => {
@@ -410,7 +411,7 @@
         size => 100
       },
       rank => {
-        data_type => "integer",
+        data_type => re(qr/^integer$/i),
         default_value => 13,
         is_nullable => 0,
         size => undef
@@ -421,7 +422,7 @@
 
   ok($schema->source("Artist")->{_columns_info_loaded} == 1, 'Columns info loaded flag set');
 
-  is_deeply (
+  cmp_deeply (
     $schema->source('Artist')->columns_info([qw/artistid rank/]),
     {
       artistid => {
@@ -431,7 +432,7 @@
         size => undef
       },
       rank => {
-        data_type => "integer",
+        data_type => re(qr/^integer$/i),
         default_value => 13,
         is_nullable => 0,
         size => undef
--- a/t/64db.t
+++ b/t/64db.t
@@ -2,6 +2,7 @@
 use warnings;
 
 use Test::More;
+use Test::Deep;
 use lib qw(t/lib);
 use DBICTest;
 
@@ -48,7 +49,7 @@
 );
 
 {
-  is_deeply (
+  cmp_deeply (
     get_storage_column_info ($schema->storage, 'artist', qw/size/),
     {
       'artistid' => {
@@ -60,7 +61,7 @@
           'is_nullable' => 1,
       },
       'rank' => {
-          'data_type' => 'integer',
+          'data_type' => re(qr/^integer$/i),
           'is_nullable' => 0,
           'default_value' => '13',
       },
